release_shared64 and release_static64 fail on debian 10 (buster)

On Debian 10 (buster) with g++ 8.3.0 I'm getting a build failure for the release_shared64 and release_static64 targets. I realize Debian isn't in BUILDING.md (only Ubuntu is), but this appears to be a difference between compilers used.

The first compilation error (actually warning, but warnings are errors in the project) is in a template and appears about 20 times, here is one instance:
```
src/next.cpp: In function ‘int next_read_packet(uint8_t*, int, void*, const int*, uint64_t*, const uint8_t*, next_replay_protection_t*)’:
src/next.cpp:1851:19: error: ‘int_value’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
             memcpy( &value, &int_value, 4 );
```

This is kind of a real issue... but only if the caller of serialize_float_internal doesn't check the return value.

After fixing that issue, two more compilation errors (again warnings) appear:

```
==== Building next (release_static64) ====
next.cpp
next_linux.cpp
src/next_linux.cpp: In function ‘int get_connection_type()’:
src/next_linux.cpp:806:20: error: ‘char* strncpy(char*, const char*, size_t)’ specified bound 16 equals destination size [-Werror=stringop-truncation]
             strncpy( req.ifr_name, i->ifa_name, IFNAMSIZ );
             ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/next_linux.cpp:835:20: error: ‘char* strncpy(char*, const char*, size_t)’ specified bound 16 equals destination size [-Werror=stringop-truncation]
             strncpy( pwrq.ifr_name, i->ifa_name, IFNAMSIZ );
             ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[1]: *** [next.make:409: obj/Static64/Release/next/next_linux.o] Error 1
make: *** [Makefile:81: next] Error 2
```

These are due to the fact strncpy won't copy a null terminator to the destination string if the source string length >= IFNAMSIZ. Probably not an issue in real life (unless you have some strangely long interface names).

# System Configuration

```
networknext-sdk (master)$ lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 10 (buster)
Release:        10
Codename:       buster
networknext-sdk (master)$ g++ --version
g++ (Debian 8.3.0-6) 8.3.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

networknext-sdk (master)$ premake5 --version
premake5 (Premake Build Script Generator) 5.0.0-alpha14
networknext-sdk (master)$ gcc --version
gcc (Debian 8.3.0-6) 8.3.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

--- a/src/next.cpp
+++ b/src/next.cpp
@@ -1846,7 +1846,7 @@ namespace next
             memcpy( &int_value, &value, 4 );
         }
         bool result = stream.SerializeBits( int_value, 32 );
-        if ( Stream::IsReading )
+        if ( Stream::IsReading && result )
         {
             memcpy( &value, &int_value, 4 );
         }

--- a/src/next_linux.cpp
+++ b/src/next_linux.cpp
@@ -803,7 +803,7 @@ static int get_connection_type()
         {
             struct ifreq req;
             memset( &req, 0, sizeof( req ) );
-            strncpy( req.ifr_name, i->ifa_name, IFNAMSIZ );
+            strncpy( req.ifr_name, i->ifa_name, IFNAMSIZ - 1 );
 
             if ( ioctl( sock, SIOCGIFINDEX, &req ) == -1 )
             {
@@ -832,7 +832,7 @@ static int get_connection_type()
 
             struct iwreq pwrq;
             memset( &pwrq, 0, sizeof( pwrq ) );
-            strncpy( pwrq.ifr_name, i->ifa_name, IFNAMSIZ );
+            strncpy( pwrq.ifr_name, i->ifa_name, IFNAMSIZ - 1 );
 
             if ( ioctl( sock, SIOCGIWNAME, &pwrq ) == -1 )
             {

